### PR TITLE
Graffiti flag

### PR DESCRIPTION
**Pull request by metanull-operator**
[Resolves] #4057 

---

# Description

Base implementation of graffiti flag allowing string to be encoded into proposed blocks.

- Added graffiti flag to validator.
- Updated data structures to allow graffiti string to be passed down to validator struct
- Added graffiti to BlockRequestin validator ProposeBlock
- Updated RPC RequestBlock proto to support passing graffiti in as string
- RequestBlock converts to 32 bytes adds to BeaconBlockBody
- Includes single test case on validator side

https://api.prylabs.net/eth/v1alpha1/beacon/blocks?slot=169564
https://www.beaconcha.in/block/169564

### CODE REVIEW

File: proto/beacon/rpc/v1/services.proto [hunk 1]
```diff
@@ -35,6 +35,7 @@ service ValidatorService {
 message BlockRequest {
   uint64 slot = 1;
   bytes randao_reveal = 2;
+  string graffiti = 3;
```
Comment (prestonvanloon) [inline proto/beacon/rpc/v1/services.proto]: cc: @rauljordan we'll need this upstream in ethereumapis. I'll send a PR

Comment (prestonvanloon) [reply proto/beacon/rpc/v1/services.proto]: Oh yeah, this should be bytes as @terencechain mentioned

**Comment by metanull-operator** [comment]
No problem. I know where that line is.

File: validator/client/validator.go [hunk 2]
```diff
@@ -29,6 +29,7 @@ type validator struct {
 	proposerClient       pb.ProposerServiceClient
 	validatorClient      pb.ValidatorServiceClient
 	attesterClient       pb.AttesterServiceClient
+	graffiti             string
```
Comment (nisdas) [inline validator/client/validator.go]: same here

**Pull request merged**